visualization/app/codeCharta/ui/codeMap/threeViewer/threeSceneService.ts

```diff
@@ -191,18 +190,49 @@ export class ThreeSceneService implements CodeMapPreRenderServiceSubscriber, Map
 
 			hoveredLabel.position.add(this.normedTransformVector)
 
+			this.toggleLineAnimation(hoveredLabel)
+
 			this.highlightedLabel = hoveredLabel
 		}
 	}
 
+	resetLineHighlight() {
+		this.highlightedLineIndex = -1
+		this.highlightedLine = null
+	}
+
 	resetLabel() {
 		if (this.highlightedLabel !== null) {
 			this.highlightedLabel.position.sub(this.normedTransformVector)
 			this.highlightedLabel.material.opacity = this.mapLabelColors.alpha
+
+			if (this.highlightedLineIndex > -1) {
```

Comment:
```suggestion
			if (this.highlightedLine) {
```

AFAIK it's impossible to have a label without a line and as such the `if` might not be required at all?

Reply:
This if handles temporary labels, due to them being tracked differently then normal ones.
I admit that the label/line implementation is a bit wanky in regards to how the array is filled (label/line instead of object, temporary labels at the end of it and maybe we should create a tech issue to fix it in the future as it is a problem spanning the whole project)